python/ray/util/ml_utils/checkpoint_manager.py

```diff
@@ -231,6 +232,33 @@ def __post_init__(self):
                 f"checkpoint_score_order must be either " f'"{MAX}" or "{MIN}".'
             )
 
+    @property
+    def checkpoint_score_attr(self) -> Optional[str]:
+        """Same as ``checkpoint_score_attr`` in ``tune.run``."""
+        if self.checkpoint_score_attribute is None:
+            return self.checkpoint_score_attribute
+        prefix = ""
+        if self.checkpoint_score_order == MIN:
+            prefix = "min-"
+        return f"{prefix}{self.checkpoint_score_attribute}"
```

Comment:
Instead of adding this here we should have `tune.execution.checkpoint_manager._CheckpointManager` accept a `CheckpointConfig` object. This will then deprecate the previous checkpoint-related arguments.

An open question is how we want to deal with class-trainable API parameters (checkpoint at end, checkpoint_freq) here. I think we can add them to the CheckpointConfig object and document their use case.

I'm happy if we do this in a follow-up, but in that case, please move this to a private method and call it e.g. `_tune_legacy_checkpoint_score_attr` so it's not exposed as part of the public API.